packages/nextjs/pages/api/relay.ts

```diff
@@ -5,16 +5,28 @@ import {
   SimulationResponse,
 } from "@flashbots/ethers-provider-bundle";
 import { ethers } from "ethers";
+import * as chains from "wagmi/chains";
+import { getNetworkConfig } from "~~/utils/scaffold-eth";
 
 export const maxDuration = 240;
-const mainnetProvider = new ethers.providers.InfuraProvider(1, "416f5398fa3d4bb389f18fd3fa5fb58c");
-const flashbotProvider = await FlashbotsBundleProvider.create(
-  mainnetProvider,
-  ethers.Wallet.createRandom(),
-  "https://relay.flashbots.net/",
-);
 
 export default async function handler(req: any, res: any) {
+  // Get network from request headers or query params
+  const networkId = req.headers["x-network-id"] || req.query.networkId;
+  const network = Number(networkId) === chains.sepolia.id ? chains.sepolia : chains.mainnet;
+  const networkConfig = getNetworkConfig(network);
+
+  const provider = new ethers.providers.JsonRpcProvider(
+    `https://eth-sepolia.g.alchemy.com/v2/${process.env.NEXT_PUBLIC_ALCHEMY_API_KEY}`,
```

Comment:
This is hardcoded to the sepolia Alchemy endpoint.  I guess it should use `networkConfig.alchemyUrl`.

Reply:
Oh nice catch! I should've looked better for this one!